src/WebJobs.Script/Binding/DocumentDBScriptBindingProvider.cs

```diff
@@ -89,25 +104,59 @@ public override Collection<Attribute> GetAttributes()
 
                 string databaseName = Context.GetMetadataValue<string>("databaseName");
                 string collectionName = Context.GetMetadataValue<string>("collectionName");
+                string connection = Context.GetMetadataValue<string>("connection");
 
-                DocumentDBAttribute attribute = null;
-                if (!string.IsNullOrEmpty(databaseName) || !string.IsNullOrEmpty(collectionName))
+                if (Context.IsTrigger)
                 {
-                    attribute = new DocumentDBAttribute(databaseName, collectionName);
+                    CosmosDBTriggerAttribute attributeTrigger = new CosmosDBTriggerAttribute(databaseName, collectionName);
+                    attributeTrigger.ConnectionString = connection;
```

Comment:
I just noticed the difference in this name -- we should change this to `ConnectionStringSetting`. I think we're a little inconsistent with the naming, but we decided at some point that using `Setting` here made it more clear that it wasn't an actual connection string, but rather an app setting that points to a connection string.

Same with `LeaseConnectionStringSetting` (anything with the `[AppSetting]` attribute, really).